include/bluetooth/smp.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_SMP_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_SMP_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum {
+	BT_SMP_ERR_PASSKEY_ENTRY_FAILED =       0x01,
+	BT_SMP_ERR_OOB_NOT_AVAIL =              0x02,
+	BT_SMP_ERR_AUTH_REQUIREMENTS =          0x03,
+	BT_SMP_ERR_CONFIRM_FAILED =             0x04,
+	BT_SMP_ERR_PAIRING_NOTSUPP =            0x05,
+	BT_SMP_ERR_ENC_KEY_SIZE =               0x06,
+	BT_SMP_ERR_CMD_NOTSUPP =                0x07,
+	BT_SMP_ERR_UNSPECIFIED =                0x08,
+	BT_SMP_ERR_REPEATED_ATTEMPTS =          0x09,
+	BT_SMP_ERR_INVALID_PARAMS =             0x0a,
+	BT_SMP_ERR_DHKEY_CHECK_FAILED =         0x0b,
+	BT_SMP_ERR_NUMERIC_COMP_FAILED =        0x0c,
+	BT_SMP_ERR_BREDR_PAIRING_IN_PROGRESS =  0x0d,
+	BT_SMP_ERR_CROSS_TRANSP_NOT_ALLOWED =   0x0e,
+
+	BT_SSP_ERR_AUTHENTICATION_FAIL =        0x81,
+	BT_SSP_ERR_PAIRING_NOTSUPP =            0x82,
+	BT_SSP_ERR_PAIRING_NOT_ALLOWED =        0x83,
+	BT_SSP_ERR_PIN_OR_KEY_MISSING =         0x84,
+	BT_SSP_ERR_UNSPECIFIED =                0x85,
+	BT_SSP_ERR_INVALID_PARAMS =             0x86,
```

Comment:
What I had in mind is that we'd have a higher level BT_PAIR_ERR_* enum that the protocol (SSP, SMP) specific error would be mapped to. I was thinking we wouldn't have to do a 1:1 mapping for everything from the start, rather do the ones that we need specific error for and use some BT_PAIR_ERR_UNSPECIFIED for the rest. However, seeing that the list isn't actually that long, perhaps we can just introduce dedicated errors from the start. Any reasons why you think this approach wouldn't be a good idea?

Reply:
I'm not a big fan of mapping Error codes to new error codes. In my experience it makes it harder to figure out the root cause when you have to go through multiple levels of error codes. And error codes are mainly used for debugging. Which is why I think this is best, the SMP error codes are kept. And the ones for SSP is mapped by name to the HCI names

Also (and I might be wrong) but it seems that SSP is the old way to do it. And that SMP should be used for newer version of BREDR.

If you still think creating BT_PAIR_ERR_* is better let me know, and I can switch to that, but I prefer this way.